langchain/chat_models/base.py

```diff
@@ -197,6 +197,14 @@ def _generate(
         generation = ChatGeneration(message=message)
         return ChatResult(generations=[generation])
 
+    def _agenerate(
```

Comment:
This will have to be an async method. Will defer to @agola11 on whether it makes sense to have it default to

```
await asyncio.get_event_loop().run_in_executor(self._generate(messages, stop=stop, run_manager=run_manager)
```

Reply:
Yeah, this needs to be async and we should not be calling blocking code in async methods. A good solution is having the default implementation for `async def _agenerate` call run_in_executor, as @vowelparrot suggested

Reply:
https://github.com/hwchase17/langchain/pull/4701
 

how does this look